catalog/resource_connection.go

```diff
@@ -0,0 +1,112 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.org/x/exp/slices"
+)
+
+// This structure contains the fields of catalog.UpdateConnection and catalog.CreateConnection
+// We need to create this because we need Owner, FullNameArg, SchemaName and CatalogName which aren't present in a single of them.
+// We also need to annotate tf:"computed" for the Owner field.
+type ConnectionInfo struct {
+	// User-provided free-form text description.
+	Comment string `json:"comment,omitempty" tf:"force_new"`
+	// The type of connection.
+	ConnectionType string `json:"connection_type" tf:"force_new"`
+	// Name of the connection.
+	Name string `json:"name"`
+	// Name of the connection.
+	NameArg string `json:"-" url:"-"`
+	// A map of key-value properties attached to the securable.
+	Options map[string]string `json:"options" tf:"sensitive"`
+	// Username of current owner of the connection.
+	Owner string `json:"owner,omitempty" tf:"force_new,suppress_diff"`
+	// An object containing map of key-value properties attached to the
+	// connection.
+	Properties map[string]string `json:"properties,omitempty" tf:"force_new"`
+	// If the connection is read only.
+	ReadOnly bool `json:"read_only,omitempty" tf:"force_new,computed"`
+}
+
+// suppress diff for sensitive options, which are not returned by the server
+func suppressSensitiveOptions(k, old, new string, d *schema.ResourceData) bool {
+	//this list will expand as other auth may have different sensitive options
+	sensitiveOptions := []string{"user", "password"}
```

Comment:
The source of truth for the list of secrets can be retrieved by the type manifest API:
https://temp.corp.databricks.com/?114480149a090e3e#QR0tqKyrycfLQ0OH3ZEqsyZOuWEwEtNMb+o7S4bdpXs=

Can we also add `personalAccessToken`, `access_token`, `client_secret`, and `OAuthPvtKey`

Reply:
done